### ISSUE TITLE
Kubectl 1.10 "create secret" missing kind and apiversion

### ISSUE THREAD

**Issue by costinm** [kubernetes/kubectl#399]
**Is this a request for help?** (If yes, you should use our troubleshooting guide and community support channels, see http://kubernetes.io/docs/troubleshooting/.):

No

**What keywords did you search in Kubernetes issues before filing this one?** (If you have found any duplicates, you should instead reply there.):

"create secret"

---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):

Bug

**Kubernetes version** (use `kubectl version`):

1.10

**Environment**:
- **Cloud provider or hardware configuration**:
- **OS** (e.g. from /etc/os-release):
- **Kernel** (e.g. `uname -a`):
- **Install tools**:
- **Others**:

**What happened**:

``` 
kubectl1.10 create secret generic foo --from-file=a.pem=... --dry-run -o yaml
data:
  a.pem: RGViaWFuIEdOVS9MaW51eCBidXN0ZXIvc2lkCg==
metadata:
  creationTimestamp: null
  name: foo
```

**What you expected to happen**:

```
kubectl1.9 create secret generic foo --from-file=a.pem=... --dry-run -o yaml
apiVersion: v1
data:
  a.pem: RGViaWFuIEdOVS9MaW51eCBidXN0ZXIvc2lkCg==
kind: Secret
metadata:
  creationTimestamp: null
  name: foo
```

**How to reproduce it** (as minimally and precisely as possible):

Run the above commands with kubectl 1.9 and 1.10, notice the difference in result.

**Anything else we need to know**:

Istio 0.7 was using the command as part of one of install scripts. We will stop using
the script 0.8, recommend users stick with kubectl 1.9 until istio 0.8 is released.

**Comment by fmoctezuma** [comment]
I think this is related with https://github.com/kubernetes/kubernetes/pull/61808

**Comment by liggitt** [comment]
yes, this is resolved in 1.10.1 in kubernetes/kubernetes#61808
